src/nvim/strings.c

```diff
@@ -455,23 +455,19 @@ char_u *vim_strchr(const char_u *const string, const int c)
   }
 }
 
-/*
- * Search for last occurrence of "c" in "string".
- * Return NULL if not found.
- * Does not handle multi-byte char for "c"!
- */
+/// strrchr() version which does not handle multi-byte char for "c"!
+///
+/// @param[in]  string  String to search in.
+/// @param[in]  c  Character to search for.
+///
+/// @return Pointer to the last occurrence in string or NULL
+///         if it was not found. NUL character is not
+///         handled, use `strlen()` instead.
 char_u *vim_strrchr(const char_u *string, int c)
   FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
 {
-  const char_u *retval = NULL;
-  const char_u *p = string;
-
-  while (*p) {
-    if (*p == c)
-      retval = p;
-    MB_PTR_ADV(p);
-  }
-  return (char_u *) retval;
+  assert(c > 0 && c < 0x80);
```

Comment:
We could allow `c==0`, because it is nowhere used. Otherwise this limitation should be mentioned in the function description.

Reply:
I tried to follow the spirit of `vim_strchr` above, and the comment currently says to use `strlen` for this scenario - maybe more as a coding style guideline?
Should `strchr` and `strrchr` be consistent about whether `c==0` is possible?